### DISCUSSION TITLE
Action naming in v3?

### DISCUSSION THREAD

**Discussion by ajnsn** [filamentphp/filament#7607] [Help] [answered]
If you have a simple action on a panel page, I noticed that the name of action and the livewire method name have some not so obvious connection. At least if you use closures.

If you do not name your action within, e.g. `Action::make('test')` like the method name, e.g. `testAction()`, the action won't fire. Moreover wrong named actions will trigger different action closures. 

We had a lot of this badly named Closure-Actions and after playing around with v3 I had a long journey to find this :) Is this something "meant to be" or am I missing something?

Thanks!

Examples:

ActionPage.php
```php 
use Filament\Actions\Action;
use Filament\Pages\Page;

class ActionPage extends Page
{
    // ...

    public function testAction(): Action
    {
        return Action::make('wrongName')
            ->label('Does not work, not named as the method testAction()')
            ->action(fn () => dd('testAction()'));
    }

    public function test2Action(): Action
    {
        return Action::make('test2')
            ->label('This works, named like the method test2Action()')
            ->action(fn () => dd('test2Action()'));
    }

    public function test3Action(): Action
    {
        return Action::make('test')
            ->label('This is named "test" and triggers the testAction() closure instead of the closure here.')
            ->action(fn () => dd('test3Action()'));
    }
}
```

action-page.blade.php
```blade
<x-filament-panels::page>
    <x-filament-panels::form>
        {{ $this->form }}

        {{ $this->testAction }}
        {{ $this->test2Action }}
        {{ $this->test3Action }}
    </x-filament-panels::form>
</x-filament-panels::page>
```

Repo: https://github.com/ajnsn/filament-action-naming

**Comment by danharrin** [comment, marked as answer]
Yeah it's meant to be. Essentially, we rely on the name of the action (`Action::make('x')`) to dictate what goes in the `wire:click`. And if we can't then locate that Action object again (by guessing the name of the method from the name of the action), then we can't run the action.
